keras_cv/models/backbone.py

```diff
@@ -0,0 +1,197 @@
+        return None
+
+    @backbone_level_outputs.setter
+    def backbone_level_outputs(self, value):
+        self._backbone_level_outputs = value
+
+    def extract_features(self, min_level=None, max_level=None):
+        """Convert the application model into a model backbone for other tasks.
+
+        The backbone model will usually take same inputs as the original
+        application model, but produce multiple outputs, one for each feature
+        level. Those outputs can be feed to network downstream, like FPN and RPN.
+
+        The output of the backbone model will be a dict with int as key and
+        tensor as value. The int key represent the level of the feature output.
+        A typical feature pyramid has five levels corresponding to scales P3,
+        P4, P5, P6, P7 in the backbone. Scale Pn represents a feature map 2n
+        times smaller in width and height than the input image.
+
+        Args:
+            min_level: optional int, the lowest level of feature to be included
+                in the output. Default to model's lowest feature level (based on
+                the model structure).
+            max_level: optional int, the highest level of feature to be included
+                in the output. Default to model's highest feature level (based
+                on the model structure).
+
+        Returns:
+            a `tf.keras.Model` which has dict as outputs.
+        Raises:
+            ValueError: When the model is lack of information for feature level,
+            and can't be converted to backbone model, or the min_level/max_level
+            param is out of range based on the model structure.
+        """
+        if self._backbone_level_outputs is not None:
+            backbone_level_outputs = self._backbone_level_outputs
+            model_levels = list(sorted(backbone_level_outputs.keys()))
+            if min_level is not None:
+                if min_level < model_levels[0]:
+                    raise ValueError(
+                        f"The min_level provided: {min_level} should be in "
+                        f"the range of {model_levels}"
+                    )
+            else:
+                min_level = model_levels[0]
+
+            if max_level is not None:
+                if max_level > model_levels[-1]:
+                    raise ValueError(
+                        f"The max_level provided: {max_level} should be in "
+                        f"the range of {model_levels}"
+                    )
+            else:
+                max_level = model_levels[-1]
+
+            outputs = {}
+            for level in range(min_level, max_level + 1):
+                outputs[level] = backbone_level_outputs[level]
+
+            return tf.keras.Model(inputs=self.inputs, outputs=outputs)
+
+        else:
+            raise ValueError(
+                "The current model doesn't have any feature level "
+                "information so extraction not possible."
```

Comment:
A helpful error message should give you pointers to steps you can take to address the problem (the purpose of error messages is to minimize debugging time when you encounter the error)

Reply:
Thanks will add this to the Issue as well!

Reply:
We can literally add a link to open a github issue perhaps?

Reply:
Filed #1447